src/test/java/dog/pawbook/logic/commands/DropCommandTest.java

```diff
@@ -0,0 +1,160 @@
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+        assertCommandSuccess(dropCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_dropManyDogsOneProgram_success() {
+        Program editedProgram = (Program) expectedModel.getEntity(ID_TWENTY);
+
+        Set<Integer> programIdSet = new HashSet<>();
+        List<Integer> dogIdList = Arrays.asList(ID_FOUR, ID_SIX, ID_EIGHT);
+        Set<Integer> dogIdSet = new HashSet<>(dogIdList);
+        programIdSet.add(ID_TWENTY);
+
+        DropCommand dropCommand = new DropCommand(dogIdSet, programIdSet);
+        String expectedMessage = String.format(DropCommand.MESSAGE_SUCCESS_FORMAT,
+                dogIdSet.stream()
+                        .map(String::valueOf)
+                        .collect(Collectors.joining(", ")),
+                ID_TWENTY);
+
+        HashSet<Integer> updateDroppedDogs = new HashSet<>(editedProgram.getDogIdSet());
+        updateDroppedDogs.removeAll(dogIdSet);
+
+        expectedModel.setEntity(ID_TWENTY, new Program(editedProgram.getName(),
+                editedProgram.getSessions(), editedProgram.getTags(), updateDroppedDogs));
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+
+        assertCommandSuccess(dropCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_dropManyDogsManyPrograms_failure() {
+        Program editedProgram1 = (Program) expectedModel.getEntity(ID_TWENTY);
+        Program editedProgram2 = (Program) expectedModel.getEntity(ID_TWENTY_ONE);
+
+        Set<Integer> dogIdSet1 = editedProgram1.getDogIdSet();
+        Set<Integer> dogIdSet2 = editedProgram2.getDogIdSet();
+        Set<Integer> dogIdSet = new HashSet<>();
+        dogIdSet.addAll(dogIdSet1);
+        dogIdSet.addAll(dogIdSet2);
+        List<Integer> programIdList = Arrays.asList(ID_TWENTY, ID_TWENTY_ONE);
+        Set<Integer> programIdSet = new HashSet<>(programIdList);
+        programIdSet.addAll(programIdList);
+
+        HashSet<Integer> updatedDroppedDogs1 = new HashSet<>(editedProgram1.getDogIdSet());
+        updatedDroppedDogs1.remove(ID_FOUR);
+
+        expectedModel.setEntity(ID_TWENTY, new Program(editedProgram1.getName(),
+                editedProgram1.getSessions(), editedProgram1.getTags(), updatedDroppedDogs1));
+
+        HashSet<Integer> updatedDroppedDogs2 = new HashSet<>(editedProgram2.getDogIdSet());
+        updatedDroppedDogs2.remove(ID_EIGHT);
+
+        expectedModel.setEntity(ID_TWENTY_ONE, new Program(editedProgram2.getName(),
+                editedProgram2.getSessions(), editedProgram2.getTags(), updatedDroppedDogs2));
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+
+        DropCommand dropCommand = new DropCommand(dogIdSet, programIdSet);
+
+        assertThrows(AssertionError.class, () -> dropCommand.execute(model));
+    }
```

Comment:
there could be more failures as well, like drop a dog that's not enrolled etc.

Reply:
haode!!